src/main/webapp/recipe-display.html

```diff
@@ -7,44 +7,91 @@
       <br />
-      <button type="button" onclick="submitComment(getRecipeID())" value="Submit">Submit</button>
+      <button type="button" id="submit-comment" value="Submit">Submit</button>
     </div>
     <div id="comments"></div>
-    <button type="button" onclick="deletePost(getRecipeID())">Delete Post</button>
+    <button type="button" id="delete-post">Delete Post</button>
 
+    <!-- Firebase App (the core Firebase SDK) is always required and must be listed first -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-app.js"></script>
+
+    <!-- If you enabled Analytics in your project, add the Firebase SDK for Analytics -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-analytics.js"></script>
+
+    <!-- Add Firebase products that you want to use -->
+    <script src="https://www.gstatic.com/firebasejs/7.16.1/firebase-auth.js"></script>
+    
     <!-- Page scripts -->
-    <script>
+    <script type="module">
+      import { app } from "/scripts/init.js"
+
+      /** On window load, initialize the page. */
+      window.onload = function () {
+        initPage()
+      }
+
+      /** Release module-scope funcs to the HTML page and load page elements. */
+      function initPage() {
+        window.getRecipeID = getRecipeID
+        window.submitComment = submitComment
+        window.getDetailedRecipe = getDetailedRecipe
+        window.getComments = getComments
+        window.deletePost = deletePost
+        
+        // Load the recipe and comments on the page.
+        getDetailedRecipe(getRecipeID())
+        getComments(getRecipeID())
+
+        // Set button click listeners.
+        document.getElementById("delete-post").addEventListener("click", deletePost)
+        document.getElementById("submit-comment").addEventListener("click", submitComment)
+      }
+
       /** Returns the recipe ID in the query string for this page. */
       function getRecipeID() {
         const params = new URLSearchParams(window.location.search)
         return params.get("recipeID")
       }
 
-      function submitComment(recipeID) {
+      /** Submit a comment -- disallow if no user is signed into the webapp. */
+      function submitComment() {
         var commentContent = document.getElementById("comment-content")
-        console.log(recipeID)
-        fetch("/api/comment?recipeID=" + recipeID, {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify({
-            content: commentContent.value,
-          }),
-        }).then((response) => {
-          if (response.status === 202) {
-            getComments(recipeID)
-          } else {
-            alert("Error " + response.status.toString())
-          }
-        })
+        var recipeID = getRecipeID
+
+        if (app.auth().currentUser) {
+          app
+            .auth()
+            .currentUser.getIdToken(/* forceRefresh */ true)
```

Comment:
Would it be possible to move this to its own function? Not all that important but once users can edit and delete comments we may want to avoid repetition.

Reply:
I could make `app.auth().currentUser.getIdToken(...)` its own function, but it seems kind of redundant -- especially since everything else in the `.then()` is unique to operation. Lmk what you think.

Reply:
I got you, no need to change anything then!